### PR TITLE
fix: [CDS-58414]: container step plan creator

### PR DESCRIPTION
**Pull request by abhinav-harness**
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger utAll`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/45918)

### CODE REVIEW

**Review by sahilhindwani-harness — DISMISSED** [review]
> Reviewed 6 of 6 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45918)* status: Some of the checks are too old. 6 file(s) to review. 1 of 2 approvals obtained.

**Review by sahilhindwani-harness — APPROVED** [review]
> Reviewed 25 of 29 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45918)* status: 30 file(s) to review. 1 of 2 approvals obtained.

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/pluginstep/InitContainerV2Step.java [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+        containerStepV2PluginProvider.getPluginsData(stepParameters, ambiance);
+    stepParameters.setPluginsData(pluginsData);
+    CIInitializeTaskParams buildSetupTaskParams =
+        containerStepInitHelper.getK8InitializeTaskParams(stepParameters, ambiance, logPrefix);
+
+    String stageId = ambiance.getStageExecutionId();
+    List<TaskSelector> taskSelectors = new ArrayList<>();
+
+    TaskData taskData = getTaskData(stepParameters, buildSetupTaskParams);
+    return StepUtils.prepareTaskRequest(ambiance, taskData, kryoSerializer, TaskCategory.DELEGATE_TASK_V2, null, true,
```
Comment (harshjain12) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/pluginstep/InitContainerV2Step.java:76]: We have changed this and now ci manager creates it's own task for init

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+            request.getAccountId(), OSType.fromString(request.getOsType()), request.getAmbiance(), 0, 0);
+    return PluginCreationResponse.newBuilder()
+        .setPluginDetails(
+            PluginDetails.newBuilder().putAllEnvVariables(pluginCompatibleStepContainerDefinition.getEnvVars()))
+        .build();
+  }
+
+  @Override
+  public boolean isSupported(String stepType) {
+    if (GIT_CLONE.getDisplayName().equals(stepType)) {
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java:65]: will we support only git clone for now ?
Can we log warning if we see non git clone step ?

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    try {
+      ciAbstractStepNode = YamlUtils.read(stepJsonNode, CIAbstractStepNode.class);
+    } catch (IOException e) {
+      throw new ContainerPluginParseException(
+          String.format("Error in parsing CI step for step type [%s]", request.getType()), e);
+    }
+    // todo(abhinav): get used ports from request
+    pluginCompatibleStep = (PluginCompatibleStep) ciAbstractStepNode.getStepSpecType();
+    Set<Integer> usedPorts = new HashSet<>();
+    PortFinder portFinder = PortFinder.builder().startingPort(PORT_STARTING_RANGE).usedPorts(usedPorts).build();
```
Comment (harshjain12) [inline 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java:49]: empty usedPorts ?

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java (hunk 2 above)
Comment (abhinav-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java:65]: yes we will change logic here

File: 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java (hunk 3 above)
Comment (abhinav-harness) [reply 332-ci-manager/service/src/main/java/io/harness/ci/execution/plugin/CiPluginStepInfoProvider.java:49]: addressed in follow up

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/pluginstep/InitContainerV2Step.java (hunk 1 above)
Comment (abhinav-harness) [reply pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/pluginstep/InitContainerV2Step.java:76]: pipeline svc is creating task for init now, i dont see the issue

**Comment by abhinav-harness** [comment]
trigger ut2

**Review by rohitk-h — APPROVED** [review]
> Reviewed 1 of 6 files at r1, 25 of 29 files at r2, 4 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45918)* status: 3. unresolved discussions.

**Comment by abhinav-harness** [comment]
trigger ut4

**Comment by abhinav-harness** [comment]
trigger messagecheck

**Review by rohitk-h — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45918)* status: 3. unresolved discussions.

**Review by yogesh-chauhan — COMMENTED** [review]
> Reviewed 1 of 6 files at r1, 15 of 29 files at r2, 1 of 4 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/45918)* status: 3. unresolved discussions.

**Review by yogesh-chauhan — APPROVED** [review]

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/pluginstep/InitContainerV2Step.java (hunk 1 above)
Comment (harshjain12) [reply pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/pluginstep/InitContainerV2Step.java:76]: queuing will not work without it

Comment (harshjain12) [reply pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/pluginstep/InitContainerV2Step.java:76]: @aman-harness

**Review by harshjain12 — APPROVED** [review]

**Comment by AnshuH** [comment]
trigger review

**Comment by abhinav-harness** [comment]
trigger reviewCheck

**Comment by abhinav-harness** [comment]
trigger review

**Pull request merged**